test/pipe.js

```diff
@@ -50,23 +50,54 @@ assert.equal(shell.error(), null);
 assert.equal(result.toString(), shell.cat('resources/uniq/pipeSorted').toString());
 
 // Synchronous exec
-// TODO: add windows tests
-if (process.platform !== 'win32') {
+if (process.platform === 'win32') {
+  // Windows-specific
+
+  // Ensure the user does not have the wrong version of FIND. I.e.,
+  // require the following to return error.
+  result = shell.exec('CD..&FIND . -name no');
+  if (!shell.error()) {
+    console.error('Warning: Cannot verify piped exec: Found POSIX-like find(1) in PATH. This test assumes a Windows environment with its FIND command (fix your PATH, exit cygwin/mingw32/MSYS2).');
+  } else {
```

Comment:
Is there a clean way to do this without a nested if-else? It makes things a little confusing to read. One preferable alternative would be an else-if ladder.

If we wait for #512 to be merged, then you might be able to simplify this to:

```js
if (!shell.error()) {
  console.error('Cannot verify...');
  return; // exit early
}
// normal case....
```

This does not have to be blocked by #512 though if we can do a similarly readable alternative.